File: leetcodeInterviewQuestions/Question409/Question409.cpp
class Solution
{
public:
      int longestPalindrome(string s)
      {
            unordered_map<char, int> u;
            for (int i = 0; i < s.length(); i++)
            {
                  u[s[i]]++;
            }

            int ans = 0;
            bool c = true;
            int maxi = INT_MIN;
            for (auto i : u)
            {
                  if (i.second % 2 == 0)
                        ans += i.second;
                  else
                  {
                        c = false;
                        maxi = max(INT_MIN, i.second);
                        ans += (maxi - 1);
                  }
            }
            if (!c)
                  ans += 1;
            return ans;
      }
};

File: leetcodeInterviewQuestions/Question409/Question409.py
from collections import Counter
class Solution:
    def longestPalindrome(self, s: str) -> int:
        c = Counter(s)
        o, oc, e = 0, 0, 0
        for i in c.keys():
            if c[i]%2 == 0:
                e += c[i]
            else:
                oc += 1
                o += c[i] - 1
        if oc >= 1:
            o += 1
        return e + o
  
  
###############

class Solution:
    def longestPalindrome(self, s: str) -> int:
        ss = set()
        for letter in s:
            if letter not in ss:
                ss.add(letter)
            else:
                ss.remove(letter)
        if len(ss) != 0:
            return len(s) - len(ss) + 1
        else:
            return len(s)
        